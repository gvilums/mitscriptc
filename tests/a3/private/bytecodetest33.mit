print_a = fun () {
  // these types of comments work too.
  print("a");
};

print_or_a = fun (something) {
  if (!(something == None)) {
    print(something);
  } else {
    // ok, fine
    print_a();
  }
};

print_astonished = fun () {
  print("!@#$%%^&&*()~`");
  //print("\n\n\n\n\t\t\t\t\"\"\"\\\\\\");
};

lut = {
    hello: "world";
    how_many: "licks does it take?";
    to_get_to: "the center of a tootsie pop?";
};

print_lut = fun (param) {
    print(lut[param]);
};

print_a();
print_astonished();
print_or_a("Hi!");
print_or_a(None);
print_lut("to_get_to");
