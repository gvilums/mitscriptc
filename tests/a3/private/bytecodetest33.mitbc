function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = ["a", None],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [print],
			instructions = 
			[
				load_global	0
				load_const	0
				call	1
				pop
				load_const	1
				return
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 1,
			local_vars = [something],
			local_ref_vars = [],
			free_vars = [],
			names = [print, print_a],
			instructions = 
			[
				load_local	0
				load_const	0
				eq
				not
				not
				if	6
				load_global	0
				load_local	0
				call	1
				pop
				goto	4
				load_global	1
				call	0
				pop
				load_const	0
				return
			]
		},
		function
		{
			functions = [],
			constants = ["!@#$%%^&&*()~`", None],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [print],
			instructions = 
			[
				load_global	0
				load_const	0
				call	1
				pop
				load_const	1
				return
			]
		},
		function
		{
			functions = [],
			constants = [None],
			parameter_count = 1,
			local_vars = [param],
			local_ref_vars = [],
			free_vars = [],
			names = [print, lut],
			instructions = 
			[
				load_global	0
				load_global	1
				load_local	0
				index_load
				call	1
				pop
				load_const	0
				return
			]
		}
	],
	constants = ["world", "licks does it take?", "the center of a tootsie pop?", "Hi!", None, "to_get_to"],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [print_a, print_or_a, print_astonished, lut, print_lut, hello, how_many, to_get_to, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	8
		load_func	1
		alloc_closure	0
		store_global	9
		load_func	2
		alloc_closure	0
		store_global	10
		load_func	3
		alloc_closure	0
		store_global	0
		load_func	4
		alloc_closure	0
		store_global	1
		load_func	5
		alloc_closure	0
		store_global	2
		alloc_record
		dup
		dup
		dup
		load_const	0
		field_store	5
		load_const	1
		field_store	6
		load_const	2
		field_store	7
		store_global	3
		load_func	6
		alloc_closure	0
		store_global	4
		load_global	0
		call	0
		pop
		load_global	2
		call	0
		pop
		load_global	1
		load_const	3
		call	1
		pop
		load_global	1
		load_const	4
		call	1
		pop
		load_global	4
		load_const	5
		call	1
		pop
	]
}