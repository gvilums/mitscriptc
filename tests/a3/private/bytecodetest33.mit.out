a
!@#$%%^&&*()~`
Hi!
a
the center of a tootsie pop?
